content/300-guides/300-upgrade-guides/200-upgrading-to-latest/10-upgrade-to-2290.mdx

```diff
@@ -0,0 +1,194 @@
+
+1.  Enable the `namedConstraints` Preview:
+
+    ```prisma highlight=3;normal
+    generator client {
+      provider        = "prisma-client-js"
+      previewFeatures = ["namedConstraints"]
+    }
+
+    datasource db {
+      provider = "postgresql"
+      url      = env("DATABASE_URL")
+    }
+
+    model User {
+      id    Int    @id() @default(autoincrement())
+      name  String @unique
+      posts Post[]
+    }
+
+    model Post {
+      id          Int     @id @default(autoincrement())
+      title       String
+      authorName  String  @default("Anonymous")
+      author      User?   @relation(fields: [authorName], references: [name])
+    }
+    ```
+
+1.  Introspect your **development database** to populate the Prisma schema with constraint and index names in your underlying database _that do not match Prisma's naming convention_:
+
+    ```terminal
+    npx prisma migrate introspect
+    ```
+
+    In this example, the highlighted constraints did not conform to Prisma's default naming convention and now include the `map` attribute field:
+
+    ```prisma highlight=11;normal
+    model User {
+      id    Int    @id(map: "Custom_Constraint_Name") @default(autoincrement())
+      name  String @unique
+      posts Post[]
+    }
+
+    model Post {
+      id    Int @id @default(autoincrement())
+      title String
+      authorName String @default("Anonymous")
+      author User? @relation(fields: [authorName], references: [name], map: "Custom_Foreign_Key_Constraint")
+    }
+    ```
+
+1.  Optionally introspect your **production database** and double-check that constraint names in production and development are in sync. If you have external applications or scripts that rely on a specific constraint name in production, make sure that these constraint names are represented in the Prisma schema.
+
+    In the following example, the `Post` model has a foreign key constraint with a custom name in production that does not match development:
+
+      <TabbedContent tabs={[<FileWithIcon text="Development environment" icon="prisma"/>, <FileWithIcon text="Production environment" icon="prisma"/>]}>
+
+      <tab>
+
+    ```prisma highlight=5;normal
+    model Post {
+      id    Int @id @default(autoincrement())
+      title String
+      authorName String @default("Anonymous")
+      author User? @relation(fields: [authorName], references: [name], map: "Custom_Foreign_Key_Constraint")
+    }
+    ```
+
+      </tab>
+
+      <tab>
+
+    ```prisma highlight=5;normal
+    model Post {
+      id    Int @id @default(autoincrement())
+      title String
+      authorName String @default("Anonymous")
+      author User? @relation(fields: [authorName], references: [name], map: "Custom_Production_Name")
+    }
+    ```
+
+      </tab>
+
+      </TabbedContent>
+
+    To synchronize constraint names between development and production, make sure that the constraint name in your Prisma schema is `map: "Custom_Production_Name"`.
```

Comment:
What does does this mean really? Sync which one with what? Wouldn't this be more "if you want to make production win and fix your dev env"?

Reply:
That's why I added the todo to have an explicit step 0 that explains how and why to check prod / dev for misalignment and fix it. I think that topic is tricky enough for its own point.